docs/dev/adding_instruments.rst

```diff
@@ -105,9 +106,48 @@ This is a minimal instrument definition:
+* If there are already similar instruments in the same "family" (like a power supply) in pymeasure, try to follow their lead where applicable. It's better if, e.g., all power supplies have a :code:`current_limit` instead of an assortment of :code:`current_max`, :code:`Ilim`, :code:`max_curr`, etc.
+* If there is already an instrument with a similar command set, check if you can inherit from that one and just tweak a couple of things. This massively reduces code duplication and maintenance effort. The section :ref:`instruments_with_similar_features` shows how to achieve that.
+* The bulk of your instrument's interface will probably be made up of properties for quantities to set and/or read out. Our custom properties (see :ref:`writing_properties` ff. below) offer some convenience features and are therefore preferable, but plain Python properties are also fine.
+* "Actions", commands or verbs should typically be methods, not properties: :code:`recall()`, :code:`trigger_scan()`, :code:`prepare_resistance_measurement()`, etc.
+* This separation between properties and methods also naturally helps with observing the `"command-query separation" principle <https://en.wikipedia.org/wiki/Command%E2%80%93query_separation>`__.
+* If your instrument has multiple identical channels, see XXX. TODO: write section on channel implementations
+
+In principle you are free to write any methods that are necessary for interacting with the instrument. When doing so, make sure to use the :code:`self.ask(command)`, :code:`self.write(command)`, and :code:`self.read()` methods to issue commands instead of calling the adapter directly. If the communication requires changes to the commands sent/received, you can override these methods in your instrument, for further information see advanced_communication_protocols_.
+
+In practice, we have developed a number of best practices for making instruments easy to write and maintain. The following sections detail these, which are highly encouraged to follow.
+
+Common instrument types
+***********************
+There are a number of categories that many instruments fit into.
+In the future, pymeasure should gain an abstraction layer based on that, see `this issue <https://github.com/pymeasure/pymeasure/issues/416>`__.
+Until that is ready, here are a couple of guidelines towards a more uniform API.
+Note that not all already available instruments follow these, but expect this to be harmonized in the future.
+
+Frequent properties
+-------------------
+If your instrument has an **output** that can be switched on and off, use a :ref:`boolean property <boolean-properties>` called :code:`output_enabled`.
+
+Power supplies
+--------------
+PSUs typically can measure the *actual* current and voltage, as well as have settings for the voltage level and the current limit.
+To keep naming clear and avoid confusion, implement the properties :code:`current`, :code:`voltage`, :code:`voltage_setpoint` and :code:`current_limit`, respectively.
```

Comment:
Here, I have started to document things I repeat again and again during reviews. I hope with this we can start shaping our instrument APIs toward homogeny before we get around to implement #416. Feel free to extend this in the future.

Reply:
Good idea

Reply:
ping @msmttchr @CasperSchippers : only so that you are aware of this existing, now.